### PR TITLE
Add waitContainer

### PR DESCRIPTION
**Pull request by kseo**

### CODE REVIEW

File: src/Docker/Client/Api.hs [hunk 1]
```diff
@@ -93,6 +94,11 @@ startContainer sopts cid = requestUnit POST $ StartContainerEndpoint sopts cid
 stopContainer :: forall m. Monad m => Timeout -> ContainerID -> DockerT m (Either DockerError ())
 stopContainer t cid = requestUnit POST $ StopContainerEndpoint t cid
 
+-- | Blocks until a container with the given 'ContainerID' stops,
+-- then returns the exit code
+waitContainer :: forall m. Monad m => ContainerID -> DockerT m (Either DockerError ())
```
Comment (denibertovic) [inline src/Docker/Client/Api.hs]: The doc string seems to suggest that this shall return an exit code but the type signature says otherwise. Do we want to return and exit code here? cc @jprider63

File: src/Docker/Client/Types.hs [hunk 2]
```diff
@@ -1422,4 +1423,3 @@ toJsonKeyVal vs getKey getVal = JSON.Object $ foldl f HM.empty vs
 -- | Helper function that return an empty dictionary "{}"
 emptyJsonObject :: JSON.Value
 emptyJsonObject = JSON.Object HM.empty
-
```
Comment (denibertovic) [inline src/Docker/Client/Types.hs]: This is style preference of mine. I like to keep an extra line at the end of each file. It's not relevant to this PR and I wont enforce it right now but we really should do a linter check of some sort...possibly even use some automated formatting so we don't have to do this manually. Anyway just a thought for the future. cc @jprider63

File: src/Docker/Client/Api.hs (hunk 1 above)
Comment (jprider63) [inline src/Docker/Client/Api.hs]: I agree that we should return the exit code. [Here](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/wait-a-container) are the relevant docs. 

Does the exit code range from 0 to 255 (inclusive)? Seems like this should return an `Int` or maybe an [ExitCode](http://hackage.haskell.org/package/base-4.9.0.0/docs/System-Exit.html#t:ExitCode).

Comment (denibertovic) [inline src/Docker/Client/Api.hs]: It's a standard Posix exit code. I agree that we should use the already existing ExitCode type here.

File: src/Docker/Client/Types.hs (hunk 2 above)
Comment (denibertovic) [inline src/Docker/Client/Types.hs]: I'm mostly using hlint as a linter and stylish-haskell for some basic auto formatting. But I don't really have a process set up or anything like that. (I do have vim plugins that I use that invoke the above tools). 

Admittedly these are 2 different things. Hlint will offer suggestion not merely based on style but on utilizing better code practices. Ie. shortening code and what not. I'm not yet sure how to automate that with Travis. Stylish-haskell on the other hand is just formatting. However, hindent seems to be more powerful in that regard. 

And I was looking into using hindent in full swing so I don't have to make decisions about styling at all: http://chrisdone.com/posts/hindent-5 It's been updated to support only one style (as opposed to the previous version). There are some things that I object to but I guess I could learn to live with them for the sake of consistency. 

So in summary I guess what I want is commit a hindent and hlint file into the repo and make sure people use that. And have travis tests somehow enforce that.

**Comment by denibertovic** [comment]
Hi @kseo just checking in about the rebase and the changes above. Do you need any help?

**Comment by kseo** [comment]
@denibertovic I updated the PR to return `ExitCode`.

**Comment by denibertovic** [comment]
LGTM. Thanks @kseo :cake:
